src/schema/decimal.rs

```diff
@@ -0,0 +1,84 @@
+use crate::utils;
+use bigdecimal::BigDecimal;
+use num_bigint::ToBigInt;
+use std::io::{Read, Write};
+use std::str::FromStr;
+use yaserde::{YaDeserialize, YaSerialize};
+
+#[derive(Default, PartialEq, Debug)]
+pub struct Decimal {
+    pub value: BigDecimal,
+}
+
+impl Decimal {
+    fn from_bigdecimal(bigdecimal: BigDecimal) -> Self {
+        Decimal{value: bigdecimal}
+    }
+
+    fn to_bigdecimal(&self) -> Option<BigDecimal> {
```

Comment:
What if we remove this `impl` at all? 
`to_bigdecimal` is unused.
`from_bigdecimal` can be replaced with calling constructor in place since `value` is pub.

If we leave this `impl` then probably need to mark fn's as `pub`. Also not sure if `Option` is needed here

Reply:
I've added `to_bigdecimal` and `from_bigdecimal` for consistency with Integer implementation. Let's keep them. Marked them as `pub`, thank you.